src/xrpld/rpc/detail/TransactionSign.cpp

```diff
@@ -710,6 +708,112 @@ transactionFormatResultImpl(Transaction::pointer tpTrans, unsigned apiVersion)
 
 //------------------------------------------------------------------------------
 
+static XRPAmount
+getTxFee(Application const& app, Config const& config, Json::Value tx)
+{
+    // autofilling only needed in this function so that the `STParsedJSONObject`
+    // parsing works properly it should not be modifying the actual `tx` object
+    if (!tx.isMember(jss::Fee))
+    {
+        tx[jss::Fee] = "0";
+    }
+
+    if (!tx.isMember(jss::Sequence))
+    {
+        tx[jss::Sequence] = "0";
+    }
+
+    if (!tx.isMember(jss::SigningPubKey))
+    {
+        tx[jss::SigningPubKey] = "";
+    }
+
+    if (!tx.isMember(jss::TxnSignature))
+    {
+        tx[jss::TxnSignature] = "";
+    }
+
+    if (tx.isMember(jss::Signers))
+    {
+        if (!tx[jss::Signers].isArray())
+            return config.FEES.reference_fee;
+
+        // check multi-signed signers
+        for (auto& signer : tx[jss::Signers])
+        {
+            if (!signer.isMember(jss::Signer) ||
+                !signer[jss::Signer].isObject())
+                return config.FEES.reference_fee;
+            if (!signer[jss::Signer].isMember(sfTxnSignature.jsonName))
+            {
+                // autofill TxnSignature
+                signer[jss::Signer][sfTxnSignature.jsonName] = "";
+            }
+        }
+    }
+
+    STParsedJSONObject parsed(std::string(jss::tx_json), tx);
+    if (!parsed.object.has_value())
+    {
+        return config.FEES.reference_fee;
+    }
+
+    try
+    {
+        STTx const& stTx = STTx(std::move(parsed.object.value()));
+        return calculateBaseFee(*app.openLedger().current(), stTx);
+    }
+    catch (std::exception& e)
+    {
+        return config.FEES.reference_fee;
+    }
+}
+
+Json::Value
+getCurrentNetworkFee(
+    Role const role,
+    Config const& config,
+    LoadFeeTrack const& feeTrack,
+    TxQ const& txQ,
+    Application const& app,
+    Json::Value& tx,
+    int mult,
+    int div)
+{
+    XRPAmount const feeDefault = getTxFee(app, config, tx);
+
+    auto ledger = app.openLedger().current();
+    // Administrative and identified endpoints are exempt from local fees.
+    XRPAmount const loadFee =
+        scaleFeeLoad(feeDefault, feeTrack, ledger->fees(), isUnlimited(role));
+    XRPAmount fee = loadFee;
+    {
+        auto const metrics = txQ.getMetrics(*ledger);
+        auto const baseFee = ledger->fees().base;
+        auto escalatedFee =
+            toDrops(metrics.openLedgerFeeLevel - FeeLevel64(1), baseFee) + 1;
+        fee = std::max(fee, escalatedFee);
+    }
+
+    auto const limit = [&]() {
```

Comment:
This lambda is unnecessary, please remove it

Reply:
Fixed in 0b818f33287094da7e46370c20269dac49b7a540